src/components/TrafimageMaps/TrafimageMaps.js

```diff
@@ -144,14 +125,19 @@ const defaultProps = {
   },
   baseLayers: null,
   popupComponents: {},
-  menuComponents: [],
   projection: 'EPSG:3857',
   layers: null,
   apiKey: null,
   history: null,
   initialState: {},
 };
 
+/**
+ * Application context.
+ * Only use the global app context for the map and the layerService.
+ */
+export const AppContext = React.createContext();
```

Comment:
It seems a bit weird to me to use redux AND context api in the same time, putting map and layerService in redux state will have the same effect..

Even if it's one line of code I will put it in another file..

Reply:
In my view it makes sense to use the context API to share an instance object with multiple components. Redux seems better suited to mutate and listen to global app state changes.

Reply:
I asked the same question to friedjoff yesterday and agreed with his answer.